src/components/Filter/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import { Body } from "@leafygreen-ui/typography";
+import Icon, { Size } from "components/Icon";
+import { fontSize, size } from "constants/tokens";
+import Toggle from "./Toggle";
+
+interface FilterProps {
+  ["data-cy"]?: string;
+  filterText: string;
+  deleteFilter: (filter: string) => void;
+}
+
+const Filter: React.FC<FilterProps> = ({
+  "data-cy": dataCy,
+  filterText,
+  deleteFilter,
+}) => {
+  const [isVisible, setIsVisible] = useState(true);
+
+  const toggleVisible = () => {
+    setIsVisible(!isVisible);
+  };
+
+  return (
+    <FilterRow data-cy={dataCy}>
+      <LeftContainer>
+        <StyledIcon
+          glyph="X"
+          onClick={() => deleteFilter(filterText)}
+          size={Size.XLarge}
+        />
+        <StyledIcon
+          glyph={isVisible ? "Visibility" : "ClosedEye"}
+          onClick={() => toggleVisible()}
+          size={Size.XLarge}
+        />
```

Comment:
If not too much trouble, it could be nice to use these icons with an LG `IconButton` to leverage the hover states they've implemented